pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -335,7 +330,7 @@ func (m *manager) GetPodTopologyHints(pod *v1.Pod) map[string][]topologymanager.
 }
 
 func (m *manager) GetAllocatableCPUs() cpuset.CPUSet {
-	return m.allocatableCPUs.Clone()
+	return m.policy.GetAllocatableCPUs(m.state).Clone()
```

Comment:
Actually it's already a copy, we need not clone again.
Does memoryManager have the same behaviour?
Do you think it is easy to add some sort of unit test about the behavior?

Reply:
> Does memoryManager have the same behaviour?

Yes, it seems like `memoryManager` also has this problem. Should I fix them in this PR or next?

> Do you think it is easy to add some sort of unit test about the behavior?

Sure. Please take a look.

Reply:
I think we should also fix `memoryManager `and change the title for this pr

Reply:
Hi @249043822, I have made a mistake. The `memoryManager` does not have the same problem here because `GetAllocatableMemory` only collects allocatable memory which does not change all the time. But allocatable cpuset in `cpuManager` changes because it excludes static cpu assignment which corresponds with new container. So we don't need to fix `memoryManager`.